fs/dafilesrv/dafilesrv.cpp

```diff
@@ -386,7 +386,29 @@ int main(int argc, const char* argv[])
     StringBuffer componentName;
 
     // NB: bare-metal dafilesrv does not have a component specific xml
-    Owned<IPropertyTree> config = loadConfiguration(defaultYaml, argv, "dafilesrv", "DAFILESRV", nullptr, nullptr);
+    Owned<IPropertyTree> extractedGlobalConfig = createPTree("dafilesrv");
+
+#ifndef _CONTAINERIZED
+    Owned<IPropertyTree> env = getHPCCEnvironment();
+    IPropertyTree* globalTracing = env->getPropTree("Software/tracing");
+    if (globalTracing != nullptr)
+        extractedGlobalConfig->addPropTree("tracing", globalTracing);
+#endif
+
+    const char* componentTag = "dafilesrv";
+    Owned<IPropertyTree> componentDefault;
+    if (defaultYaml)
+    {
+        Owned<IPropertyTree> defaultConfig = createPTreeFromYAMLString(defaultYaml, 0, ptr_ignoreWhiteSpace, nullptr);
+        componentDefault.set(defaultConfig->queryPropTree(componentTag));
+        if (!componentDefault)
+            throw makeStringExceptionV(99, "Default configuration does not contain the tag %s", componentTag);
+    }
+    else
+        componentDefault.setown(createPTree(componentTag));
```

Comment:
I think the above would be better simplified to:
```
    const char* componentTag = "dafilesrv";
    Owned<IPropertyTree> defaultConfig = createPTreeFromYAMLString(defaultYaml, 0, ptr_ignoreWhiteSpace, nullptr);
    Owned<IPropertyTree> componentDefault = defaultConfig->getPropTree(componentTag);
```

or I might be tempted to encapsulate this pattern into a another loadConfiguration variety (that calls onto loadConfiguration(IPropertyTree * defaultConfig, IPropertyTree * globalConfig, ...).